### PR TITLE
New PFEM-DEM elements and changes in strategy

### PR DESCRIPTION
**Pull request by AFranci**
New elements for PFEM-DEM coupling  (derived from PFEM fluid element) and some changes in strategy to improve robustness (intermedium check of convergence).

### CODE REVIEW

**Review by jgonzalezusua — CHANGES_REQUESTED** [review]
Could you wait until we merge our branch (coupling-sdem-pfem) into master? In this way you can directly add your changes into the new python Pfem-dem coupling files.

**Comment by loumalouomega** [comment]
@AFranci  do you still need the branch?

**Pull request closed without merging**
